### PR TITLE
FROG-38: Input bad state

### PR DESCRIPTION
**Pull request by devkav**
## Description

- Update Input regex to only allow alphanumeric, dots, dashes, and underscores.

## Ticket

FROG-38 - [LINK](https://nmd2117.atlassian.net/browse/FROG-38)

### CODE REVIEW

**Comment by NicholasDawson** [comment]
I would like to see more native usage here, I think it would be more simple and efficient.

There are `:valid` and `:invalid` and many other psuedo-classes for css. That will be useful.

The native html [input](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input) element has a [type](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#input_types) attribute and a [pattern](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#pattern) attribute which I think will give us everything we need.

**Comment by devkav** [comment]
> I would like to see more native usage here, I think it would be more simple and efficient.
> 
> There are `:valid` and `:invalid` and many other psuedo-classes for css. That will be useful.
> 
> The native html [input](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input) element has a [type](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#input_types) attribute and a [pattern](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#pattern) attribute which I think will give us everything we need.



> I would like to see more native usage here, I think it would be more simple and efficient.
> 
> There are `:valid` and `:invalid` and many other psuedo-classes for css. That will be useful.
> 
> The native html [input](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input) element has a [type](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#input_types) attribute and a [pattern](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#pattern) attribute which I think will give us everything we need.

The `:valid` and `:invalid` cannot be set. They are an indicator of if they match the given pattern. This is not the use case that we want. We want valid and invalid to be determined by the parent. A text may be allowed by the pattern, but we may still not want to mark it as valid.

`pattern` does not restrict which characters are allowed, it only determines `valid`. The regex restricts the user to only enter valid chars.

I think the `type` property will be helpful for keyboards on mobile.

File: src/Input/Input.tsx [hunk 1]
```diff
@@ -162,7 +162,7 @@ const Input = ({
           value={value}
           placeholder={placeholder}
           autoComplete="off"
-          type={type === 'password' ? 'password' : 'text'}
+          type={type}
```
Comment (NicholasDawson) [inline src/Input/Input.tsx:165]: Like this, then we can pass through the string for native html type attr

File: src/Input/Input.tsx
Comment (NicholasDawson) [inline src/Input/Input.tsx:1]: lgtm

**Review by NicholasDawson — APPROVED** [review]
lgtm 😁👍

**Pull request merged**